kong/templates/nginx_kong.lua

```diff
@@ -480,6 +488,263 @@ server {
+                local code = tonumber(ngx.var.code)
+                if not code then
+                    return ngx.exit(ngx.HTTP_NOT_FOUND)
+                end
+                ngx.status = code
+                return mu.send_default_json_response({
+                  code = code,
+                })
+            }
+        }
+
+        location ~ "^/stream/(?<num>\d+)$" {
+            content_by_lua_block {
+                local mu  = require "spec.fixtures.mock_upstream"
+                local rep = tonumber(ngx.var.num)
+                local res = require("cjson").encode(mu.get_default_json_response())
+
+                ngx.header["X-Powered-By"] = "mock_upstream"
+                ngx.header["Content-Type"] = "application/json"
+
+                for i = 1, rep do
+                  ngx.say(res)
+                end
+            }
+        }
+
+        location ~ "^/post_log/(?<logname>[a-z0-9_]+)$" {
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.store_log(ngx.var.logname)
+            }
+        }
+
+        location ~ "^/post_auth_log/(?<logname>[a-z0-9_]+)/(?<username>[a-zA-Z0-9_]+)/(?<password>.+)$" {
+            access_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.filter_access_by_basic_auth(ngx.var.username,
+                                                      ngx.var.password)
+            }
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.store_log(ngx.var.logname)
+            }
+        }
+
+        location ~ "^/read_log/(?<logname>[a-z0-9_]+)$" {
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.retrieve_log(ngx.var.logname)
+            }
+        }
+
+        location ~ "^/count_log/(?<logname>[a-z0-9_]+)$" {
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.count_log(ngx.var.logname)
+            }
+        }
+
+        location ~ "^/reset_log/(?<logname>[a-z0-9_]+)$" {
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.reset_log(ngx.var.logname)
+            }
+        }
+
+        location = /echo_sni {
+            return 200 'SNI=$ssl_server_name\n';
+        }
+
+        location = /ocsp {
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.handle_ocsp()
+            }
+        }
+
+        location = /set_ocsp {
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                return mu.set_ocsp(ngx.var.arg_status)
+            }
+        }
+    }
+> end -- role ~= "data_plane"
+
+    include '*.http_mock';
```

Comment:
I mean this.

Reply:
moved to spec/fixtures/template_inject/nginx_kong_test_custom_inject.lua